### ISSUE TITLE
[BUG] - Updating hostname causes the instance to reinstall without warning

### ISSUE THREAD

**Issue by egeexyz** [vultr/terraform-provider-vultr#224]
**Describe the bug**
Updating an existing server's hostname causes Vultr to reinstall a/the OS on the server but the Terraform plan does not in any way indicate this destructive procedure (the plan actually says "will be updated in-place" instead of "force replacement"

**To Reproduce**
1. Create a vultr_instance
2. Update the host name
3. Review the plan
4. Apply the plan

**Expected behavior**
The plan should reflect the actual change the will occur (or provide a warning, such as when `enable_private_network` is enabled)

**Additional context**

What makes this issue worse is that the provider docs do not mention this _feature_ anywhere. And the example provided includes the hostname despite it being marked as _optional_. At the very least, the Terraform docs should mention which properties are destructive even outside of the Terraform plan.

**Comment by ddymko** [comment]
@egee-irl 

We have updated the docs to be more clear that changing the hostname will reinstall the OS.

You mentioned
> the plan actually says "will be updated in-place" instead of "force replacement"

This is where things get a bit tricky. The update is "in place" it's modifying the current resource all of the IP information etc stays the same. Despite the core platform doing a reinstall.

What we could do is specify `forceNew` on the hostname field but this would then have Terraform delete that instance and spin up a new one. This will result in IPs changing as it's a brand new instance which also has its downsides.

What I did was update the docs and description in #225 to be more explicit. I'll also look into having terraform log warnings whenever a plan or apply (before confirming) that hostname will result in a reinstall.
